connectors/sources/outlook.py

```diff
@@ -0,0 +1,1006 @@
+            ),
+            "type": "Task",
+            "title": task.subject,
+            "owner": task.owner,
+            "start_date": ews_format_to_datetime(
+                source_datetime=task.start_date, timezone=timezone
+            ),
+            "due_date": ews_format_to_datetime(
+                source_datetime=task.due_date, timezone=timezone
+            ),
+            "complete_date": ews_format_to_datetime(
+                source_datetime=task.complete_date, timezone=timezone
+            ),
+            "categories": [category for category in (task.categories or [])],
+            "importance": task.importance,
+            "content": task.text_body,
+            "status": task.status,
+        }
+
+    def contact_doc_formatter(self, contact, timezone):
+        return {
+            "_id": contact.id,
+            "type": "Contact",
+            "_timestamp": ews_format_to_datetime(
+                source_datetime=contact.last_modified_time, timezone=timezone
+            ),
+            "name": contact.display_name,
+            "email_addresses": [
+                email.email for email in (contact.email_addresses or [])
+            ],
+            "contact_numbers": [
+                number.phone_number
+                for number in (contact.phone_numbers)
+                if number.phone_number
+            ],
+            "company_name": contact.company_name,
+            "birthday": ews_format_to_datetime(
+                source_datetime=contact.birthday, timezone=timezone
+            ),
+        }
+
+    def attachment_doc_formatter(self, attachment, timezone):
+        return {
+            "_id": attachment.attachment_id.id,
+            "title": attachment.name,
+            "type": "attachment",
+            "_timestamp": ews_format_to_datetime(
+                source_datetime=attachment.last_modified_time, timezone=timezone
+            ),
+            "size": attachment.size,
+        }
+
+
+class OutlookClient:
+    """Outlook client to handle API calls made to Outlook"""
+
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self.configuration = configuration
+        self._logger = logger
+        self.is_cloud = self.configuration["data_source"] == OUTLOOK_CLOUD
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+
+    @cached_property
+    def _get_user_instance(self):
+        if self.is_cloud:
+            return Office365Users(
+                client_id=self.configuration["client_id"],
+                client_secret=self.configuration["client_secret"],
+                tenant_id=self.configuration["tenant_id"],
+            )
+
+        self.ssl_enabled = self.configuration["ssl_enabled"]
+        self.certificate = self.configuration["ssl_ca"]
+
+        if self.ssl_enabled and self.certificate:
+            ssl_ca = get_pem_format(self.certificate)
+        else:
+            ssl_ca = ""
```

Comment:
Shouldn't this part be done in `__init__`?

Reply:
This only required when `data_source` is server and so i kept it in this methods because `data_source` bifurcation is happening in this method

Reply:
Yes, we can also do that 👍